lib/gosu/constants.rb

```diff
@@ -0,0 +1,332 @@
+    "GP_0_DPAD_DOWN",
+    "GP_0_LEFT",
+    "GP_0_RIGHT",
+    "GP_0_UP",
+    "GP_0_DOWN",
+    "GP_0_BUTTON_0",
+    "GP_0_BUTTON_1",
+    "GP_0_BUTTON_2",
+    "GP_0_BUTTON_3",
+    "GP_0_BUTTON_4",
+    "GP_0_BUTTON_5",
+    "GP_0_BUTTON_6",
+    "GP_0_BUTTON_7",
+    "GP_0_BUTTON_8",
+    "GP_0_BUTTON_9",
+    "GP_0_BUTTON_10",
+    "GP_0_BUTTON_11",
+    "GP_0_BUTTON_12",
+    "GP_0_BUTTON_13",
+    "GP_0_BUTTON_14",
+    "GP_0_BUTTON_15",
+
+    "GP_1_DPAD_LEFT",
+    "GP_1_DPAD_RIGHT",
+    "GP_1_DPAD_UP",
+    "GP_1_DPAD_DOWN",
+    "GP_1_LEFT",
+    "GP_1_RIGHT",
+    "GP_1_UP",
+    "GP_1_DOWN",
+    "GP_1_BUTTON_0",
+    "GP_1_BUTTON_1",
+    "GP_1_BUTTON_2",
+    "GP_1_BUTTON_3",
+    "GP_1_BUTTON_4",
+    "GP_1_BUTTON_5",
+    "GP_1_BUTTON_6",
+    "GP_1_BUTTON_7",
+    "GP_1_BUTTON_8",
+    "GP_1_BUTTON_9",
+    "GP_1_BUTTON_10",
+    "GP_1_BUTTON_11",
+    "GP_1_BUTTON_12",
+    "GP_1_BUTTON_13",
+    "GP_1_BUTTON_14",
+    "GP_1_BUTTON_15",
+
+    "GP_2_DPAD_LEFT",
+    "GP_2_DPAD_RIGHT",
+    "GP_2_DPAD_UP",
+    "GP_2_DPAD_DOWN",
+    "GP_2_LEFT",
+    "GP_2_RIGHT",
+    "GP_2_UP",
+    "GP_2_DOWN",
+    "GP_2_BUTTON_0",
+    "GP_2_BUTTON_1",
+    "GP_2_BUTTON_2",
+    "GP_2_BUTTON_3",
+    "GP_2_BUTTON_4",
+    "GP_2_BUTTON_5",
+    "GP_2_BUTTON_6",
+    "GP_2_BUTTON_7",
+    "GP_2_BUTTON_8",
+    "GP_2_BUTTON_9",
+    "GP_2_BUTTON_10",
+    "GP_2_BUTTON_11",
+    "GP_2_BUTTON_12",
+    "GP_2_BUTTON_13",
+    "GP_2_BUTTON_14",
+    "GP_2_BUTTON_15",
+
+    "GP_3_DPAD_LEFT",
+    "GP_3_DPAD_RIGHT",
+    "GP_3_DPAD_UP",
+    "GP_3_DPAD_DOWN",
+    "GP_3_LEFT",
+    "GP_3_RIGHT",
+    "GP_3_UP",
+    "GP_3_DOWN",
+    "GP_3_BUTTON_0",
+    "GP_3_BUTTON_1",
+    "GP_3_BUTTON_2",
+    "GP_3_BUTTON_3",
+    "GP_3_BUTTON_4",
+    "GP_3_BUTTON_5",
+    "GP_3_BUTTON_6",
+    "GP_3_BUTTON_7",
+    "GP_3_BUTTON_8",
+    "GP_3_BUTTON_9",
+    "GP_3_BUTTON_10",
+    "GP_3_BUTTON_11",
+    "GP_3_BUTTON_12",
+    "GP_3_BUTTON_13",
+    "GP_3_BUTTON_14",
+    "GP_3_BUTTON_15",
+
+    "GP_LEFT_STICK_X_AXIS",
+    "GP_LEFT_STICK_Y_AXIS",
+    "GP_RIGHT_STICK_X_AXIS",
+    "GP_RIGHT_STICK_Y_AXIS",
+    "GP_LEFT_TRIGGER_AXIS",
+    "GP_RIGHT_TRIGGER_AXIS",
+
+    "GP_0_LEFT_STICK_X_AXIS",
+    "GP_0_LEFT_STICK_Y_AXIS",
+    "GP_0_RIGHT_STICK_X_AXIS",
+    "GP_0_RIGHT_STICK_Y_AXIS",
+    "GP_0_LEFT_TRIGGER_AXIS",
+    "GP_0_RIGHT_TRIGGER_AXIS",
+
+    "GP_1_LEFT_STICK_X_AXIS",
+    "GP_1_LEFT_STICK_Y_AXIS",
+    "GP_1_RIGHT_STICK_X_AXIS",
+    "GP_1_RIGHT_STICK_Y_AXIS",
+    "GP_1_LEFT_TRIGGER_AXIS",
+    "GP_1_RIGHT_TRIGGER_AXIS",
+
+    "GP_2_LEFT_STICK_X_AXIS",
+    "GP_2_LEFT_STICK_Y_AXIS",
+    "GP_2_RIGHT_STICK_X_AXIS",
+    "GP_2_RIGHT_STICK_Y_AXIS",
+    "GP_2_LEFT_TRIGGER_AXIS",
+    "GP_2_RIGHT_TRIGGER_AXIS",
+
+    "GP_3_LEFT_STICK_X_AXIS",
+    "GP_3_LEFT_STICK_Y_AXIS",
+    "GP_3_RIGHT_STICK_X_AXIS",
+    "GP_3_RIGHT_STICK_Y_AXIS",
+    "GP_3_LEFT_TRIGGER_AXIS",
+    "GP_3_RIGHT_TRIGGER_AXIS",
+  ]
+
+  constants = constants + [
```

Comment:
Is there any reason to split them in two arrays?
And why not just

```suggestion
  ] +
  [
```

or
```suggestion
  ]

  constants += [
```